File: top-interview-questions-medium/Array and Strings/group-anagrams/49.py
class Solution:
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        map = {}
        for str in strs:
            lstr = list(str)
            lstr.sort()
            s = ''.join(lstr)
            map[s] = map[s] + [str] if s in map else [str]
        return list(map.values())

File: top-interview-questions-medium/Array and Strings/group-anagrams/49.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
  const map = {}
  const len = strs.length
  for (let i = 0; i < len; i++) {
    let s = strs[i].split('').sort().join()
    map[s] = map[s] ? [...map[s], strs[i]] : [strs[i]]
  }
  return Object.values(map)
}
